src/modules/esl-carousel/plugin/touch/esl-carousel.touch.mixin.ts

```diff
@@ -1,32 +1,56 @@
 import {ExportNs} from '../../../esl-utils/environment/export-ns';
-import {attr, prop, listen} from '../../../esl-utils/decorators';
-import {getTouchPoint, isMouseEvent, isTouchEvent} from '../../../esl-utils/dom';
-import {ESLMediaQuery} from '../../../esl-media-query/core';
+import {attr, prop, listen, memoize} from '../../../esl-utils/decorators';
+import {
+  ESLSwipeGestureEvent,
+  ESLSwipeGestureTarget,
+  getTouchPoint,
+  isMouseEvent,
+  isTouchEvent
+} from '../../../esl-utils/dom';
+import {ESLMediaRuleList} from '../../../esl-media-query/core';
 
 import {ESLCarouselPlugin} from '../esl-carousel.plugin';
 
 import type {Point} from '../../../esl-utils/dom';
 
+export type TouchType = 'drag' | 'swipe' | 'none';
+
+const toTouchType = (value: string): TouchType => {
+  value = value.toLowerCase();
+  if (value === ESLCarouselTouchMixin.DRAG_TYPE || value === ESLCarouselTouchMixin.SWIPE_TYPE) return value;
+  return 'none';
```

Comment:
This version would be easier to understand
```suggestion
  const types = [ESLCarouselTouchMixin.DRAG_TYPE, ESLCarouselTouchMixin.SWIPE_TYPE];
  return types.includes(value) ? value : 'none';
```

Reply:
It's better to keep it simple for now. Moreover, this is a temporary parser, as we need a generic one for "enums."